### PR TITLE
feat/create an api to signup new users

### PR DESCRIPTION
**Pull request by Alao-Abiodun**
An API to create a new user using the javascript methods without the database which means it's not persistent yet.

### CODE REVIEW

File: src/controllers/user.controller.js [hunk 1]
```diff
@@ -0,0 +1,18 @@
+const userServices = require('../services/user.services');
+const customError = require('../utils/customError');
```
Comment (kenzdozz) [inline src/controllers/user.controller.js]: Remove import if not used

File: src/utils/errorHandlers.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+const errorHandler = (req, res, err) => {
+    if (err.statusCode) {
+        res.status(statusCode).json({
+            status: 'error',
```
Comment (kenzdozz) [inline src/utils/errorHandlers.js:5]: status should bear the status code

**Review by kenzdozz — CHANGES_REQUESTED** [review]
You need to write tests

File: src/controllers/user.controller.js (hunk 1 above)
Comment (Alao-Abiodun) [reply src/controllers/user.controller.js]: I have fixed the import and adding to catch error when any field is not given

Comment (Alao-Abiodun) [reply src/controllers/user.controller.js]: I also added a test with it

File: src/utils/errorHandlers.js (hunk 2 above)
Comment (Alao-Abiodun) [reply src/utils/errorHandlers.js:5]: Yes the statusCode is a user given error code, that is if I inserted the error code myself, and other follows like that and the last one is generated by the server

File: src/test/services.test.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+                lastname: 'illyas',
+                othername: 'ayinla',
+                email: 'fawaz@gmail.com',
+                phoneNumber: '090232434542',
+                username: 'fawily',
+                registered: true, 
+                isAdmin: false
+            };
+            chai.request(app)
+                .post('/api/v1/user')
```
Comment (Kennetonah) [inline src/test/services.test.js:29]: Add test for get request

File: src/test/services.test.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+
+let should = chai.should();
+
+chai.use(chaiHttp);
+
+describe('User', () => {
+    describe('create a new User', () => {
+        it('should add user into the Arrays of object', done => {
+            // let userLength = users.length;
+            // let lastUserId = userLength[userLength - 1].id;
```
Comment (Kennetonah) [inline src/test/services.test.js:15]: Remove this comments

**Review by Kennetonah — CHANGES_REQUESTED** [review]

**Pull request merged**